### ISSUE TITLE
8192 characters limitation of a =PY() cell ?

### ISSUE THREAD

**Issue by stonebig** [microsoft/python-in-excel#62]
If i want to put a big class in a =Py() cell,  the 8192 limitation is a real pain.
I have to:
- strip all comments,
- go to 1 character indentation,
- ... etc ...etc.

In standard excel doing  a concatenation of strings, the limit was rather 32768 characters.

Could we get that 4x breathing space?

**Comment by stonebig** [comment]
Toy exercise to explain life to the product manager: fit that in one =PY() cell:
https://github.com/stonebig/baresql/blob/master/baresql/baresql.py

**Comment by stonebig** [comment]
Being able to say two cells , or only  a range of cells, defines the one =PY() is a an acceptable ugly workaround

**Comment by stonebig** [comment]
=PY(free_me!A1:A9999)

**Comment by stonebig** [comment]
Also, managing a 300 python script, it easier to have it on excel lines, in another sheet, then having to move down than up the height of the editing pane

**Comment by salabim** [comment]
I have made it possible to run arbitrarily big code from a sheet.
You can even have (one source file) modules on a sheet, which you can import as usual.

See https://www.python-in-excel.com/code/posts/ for details.

Please let me know what you (and others) think of this.

For those who use Anaconda Code (recommended!), please refer to www.github.com/salabim/anacondacode.

**Comment by stonebig** [comment]
> I have made it possible to run arbitrarily big code from a sheet. You can even have (one source file) modules on a sheet, which you can import as usual.
> 
> See https://www.python-in-excel.com/code/posts/ for details.
> 
> Please let me know what you (and others) think of this.
> 
> For those who use Anaconda Code (recommended!), please refer to [www.github.com/salabim/anacondacode](http://www.github.com/salabim/anacondacode).

I agree you can play like that... desperate situation implies desesperate measures.

In the mean time, I could rely on the new PY_thon International standard of 1 character indentation.

that makes two possible ugly workarounds, choose your poison.

**Comment by salabim** [comment]
What's ugly about my solution of putting code on a sheet?

**Comment by stonebig** [comment]
It's  versus it being it directly in the Microsoft packages.
... maintaining 100 lines of user code in a single cell is really an ergonomic catastrophy, I will try your workaround

**Comment by stonebig** [comment]
So it doesn't work exactly as hoped but nevertheless it improves my Feng shui as much as needed.

Many Thanks @salabim

**Comment by salabim** [comment]
So what did hope for that's not provided.
And what is (a) Feng shut?

**Comment by stonebig** [comment]
feng shui = way better looking excel sheet:
- pre_init code in a first hidden sheet1
- user =PY() cell reduced to 4 lines , as the [SQL] code is now external

not as expected: 
- a code like `xl("my_code!F1:F999")` cannot be in the code loaded per your method
- did transform the bottom of your code as a function `def load_code(source):`, to call it in user_code via  `load_code(xl("my_code!F1:F999"))`
